File: src/main/java/thecudster/sre/core/Changelog.java
package thecudster.sre.core;

import net.minecraft.util.EnumChatFormatting;
import thecudster.sre.util.Utils;

import java.util.ArrayList;

public class Changelog {
    private static ArrayList<ChangelogEntry> types;
    public static void init() {
        // v1.0-pre1
        types.add(new ChangelogEntry("", "1.0-pre1", ChangelogType.CHANGE));
        types.add(new ChangelogEntry("", "1.0-pre1", ChangelogType.ADDITION));
        types.add(new ChangelogEntry("", "1.0-pre1", ChangelogType.REMOVED));
    }
    public static void printChangelog(String version) {
        Utils.sendMsg(EnumChatFormatting.GREEN + "You just got the new version " + version + "! Here's what's new:");
        for (ChangelogEntry entry : types) {
            if (entry.getVersion().equals(version)) {
                Utils.sendMsg(entry.getFormattedString());
            }
        }
        Utils.sendMsg(EnumChatFormatting.RED + "❤ " + EnumChatFormatting.GOLD + "Thanks for using SRE!");
    }
}

File: src/main/kotlin/thecudster/sre/core/Changelog.kt
package thecudster.sre.core

import thecudster.sre.util.Utils.sendMsg
import thecudster.sre.core.ChangelogEntry
import thecudster.sre.core.Changelog
import thecudster.sre.core.ChangelogType
import net.minecraft.util.EnumChatFormatting
import java.util.ArrayList

object Changelog {
    private val types: ArrayList<ChangelogEntry>? = null
    fun init() {
        // v1.0-pre1
        types!!.add(ChangelogEntry("", "1.0-pre1", ChangelogType.CHANGE))
        types.add(ChangelogEntry("", "1.0-pre1", ChangelogType.ADDITION))
        types.add(ChangelogEntry("", "1.0-pre1", ChangelogType.REMOVED))
    }

    fun printChangelog(version: String) {
        sendMsg(EnumChatFormatting.GREEN.toString() + "You just got the new version " + version + "! Here's what's new:")
        for (entry in types!!) {
            if (entry.version == version) {
                sendMsg(entry.formattedString)
            }
        }
        sendMsg(EnumChatFormatting.RED.toString() + "❤ " + EnumChatFormatting.GOLD + "Thanks for using SRE!")
    }
}